doc/docs.md

```diff
@@ -1331,6 +1331,45 @@ println(numbers) // [1, 2, 3]
 ```
 When an identifier is just a single underscore, it is ignored.
 
+##### Custom iterators
+Types that implement a `next` method returning an `Option` can be iterated
+with a `for` loop. 
+
+```v
+struct SquareIterator {
+	arr []int
+mut:
+	idx int
+}
+
+fn (mut iter SquareIterator) next() ?int {
+	if iter.idx >= iter.arr.len {
+		return error('')
+	}
+
+	item := iter.arr[iter.idx]
+	iter.idx++
+	return item * item
```

Comment:
It would be nice to showcase `defer` here too.

```v
defer { iter.idx++ }
return iter.arr[iter.idx] * iter.arr[iter.idx]
```

Reply:
good idea.